common/src/main/java/com/microsoft/identity/common/internal/dto/PrimaryRefreshTokenRecord.java

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.dto;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+import lombok.EqualsAndHashCode;
+
+import static com.microsoft.identity.common.internal.dto.PrimaryRefreshTokenRecord.SerializedNames.*;
+
+@EqualsAndHashCode(callSuper = true, doNotUseGetters = true)
+public class PrimaryRefreshTokenRecord extends Credential {
+
+    public static class SerializedNames extends Credential.SerializedNames {
+        /**
+         * String of family id.
+         */
+        public static final String FAMILY_ID = "family_id";
+
+        /**
+         * String of session_key.
+         */
+        public static final String SESSION_KEY = "session_key";
+
+        /**
+         * String of session_key_rolling_date.
+         */
+        public static final String SESSION_KEY_ROLLING_DATE = "session_key_rolling_date";
+
+        /**
+         * String of prt_protocol_version.
+         */
+        public static final String PRT_PROTOCOL_VERSION = "prt_protocol_version";
+    }
+
+    /**
+     * 1st Party Application Family ID.
+     */
+    @SerializedName(FAMILY_ID)
+    private String mFamilyId;
+
+    /**
+     * PRT expiry time. This value is returned from the server as refresh_token_expires_in that
+     * should be added to local time. Measured in milliseconds from epoch (1970).
+     */
+    @SerializedName(EXPIRES_ON)
+    private String mExpiresOn;
+
+    /**
+     * Session key of PRT.
+     */
+    @SerializedName(SESSION_KEY)
+    private String mSessionKey;
```

Comment:
Can we go ahead and adopt my change to indicate whether this session key value is a literal session key or a key alias by adding a column here that could say this thing?

Reply:
It's kind-of less important because the broker should never be able to read this cache and <->.  I'd prefer to handle these keys as an interface that doesn't ever expose the key material to a cavalier user rather than as the key itself, but that detail will be in the cpp code review.

Reply:
> Can we go ahead and adopt my change to indicate whether this session key value is a literal session key or a key alias by adding a column here that could say this thing?

@AdamBJohnsonx @oldalton and I have signed off on this PR, assuming you have no other outstanding design items I am of the impression we can consider this design approved should you want to move forward with it
https://identitydivision.visualstudio.com/DevEx/_git/AuthLibrariesApiReview/pullrequest/2654

Reply:
I've added a sample to the doc as requested

Reply:
I don't see anything to add other than that unless somebody wants to hate the name.  I've marked it as complete, then.

Reply:
Thanks @AdamBJohnsonx 

@richard1122 Overall your changes LGTM, as Adam is going to be building some of his implementation off of this work, please review his proposal above and determine if it makes sense to include in this work. Apart from this, I am prepared to sign off on this implementation

Reply:
Thanks @iambmelt , @AdamBJohnsonx . I suggest we take the hardware-backed session key into another new PR. Let's discuss in [my PR in MSAL-CPP](https://github.com/AzureAD/microsoft-authentication-library-for-cpp/pull/1836). See if we can move some encryption/decryption methods into commons. (I'd like to open another PR if needed)